index.js

```diff
@@ -88,18 +88,19 @@ class ApiFactory {
 
 			let proxyPath = '/{proxy+}';
 			// default to defined path when proxy is not specified.
-			if (event.resource !== proxyPath) {
+			if (event.resource.lastIndexOf(proxyPath) === -1) {
 				if (mainEventHandler && mainEventHandler[event.resource]) {
 					definedRoute = mainEventHandler[event.resource];
 				} else if (anyEventHandler && anyEventHandler[event.resource]) {
 					definedRoute = anyEventHandler[event.resource];
 				}
 			} else {
+				// modify path to strip out potential stage in path 
+				event.path = event.resource.replace(proxyPath,'/'+ event.pathParameters.proxy)
```

Comment:
And I just checked this, not sure how conclusive this should be, but since we always know that the `/{proxy+}` is at the end, it is faster to cut of this code rather than replace: https://jsperf.com/replace-versus-slice

What do you think?

Reply:
Hmm I don't mind making the change but I think using replace makes the coder easier to reason about

Reply:
No doubt agree, but for something that literally gets executed on every single lambda call, performance is ultra critical.

Reply:
Ok ill make the change